zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -434,6 +435,25 @@ def test_virsh_audit_action(self):
                     "The action failed: {}".format(action.data["message"]))
 
 
+class NovaComputeNvidiaVgpuTest(test_utils.OpenStackBaseTest):
+    """Run nova-compute-nvidia-vgpu specific tests."""
+
+    def test_vgpu_in_nova_conf(self):
+        """Test that nova.conf contains vGPU-related settings."""
+        for unit in zaza.model.get_units('nova-compute',
+                                         model_name=self.model_name):
+            nova_conf_file = '/etc/nova/nova.conf'
+            nova_conf = str(generic_utils.get_file_contents(unit,
+                                                            nova_conf_file))
+
+            # See
+            # https://docs.openstack.org/nova/queens/admin/virtual-gpu.html
+            # https://docs.openstack.org/nova/ussuri/admin/virtual-gpu.html
+            # https://docs.openstack.org/releasenotes/nova/xena.html#deprecation-notes
+            self.assertTrue(('enabled_vgpu_types' in nova_conf) or
+                            ('enabled_mdev_types' in nova_conf))
+
+
```

Comment:
Out of curiosity what is the end result of the deployed bundles for the nova-compute-nvidia-vgpu charm? I assume there isn't a usable vgpu? If not we might consider testing the rest of the conf that gets written according to https://docs.openstack.org/nova/ussuri/admin/virtual-gpu.html etc.